aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -779,9 +779,12 @@ module aptos_framework::coin {
             burn(coin_to_burn, burn_cap);
         };
         if (fa_amount_to_burn > 0) {
-            fungible_asset::burn_from(
+            fungible_asset::address_burn_from_for_gas(
```

Comment:
burn_from is a public function that might be called from anywhere? we would need separate public(friend) for_gas variant and leave the public to emit regular events?

Reply:
address_burn_from_for_gas is the separate friend function.
Now burn function does not emit events.

Reply:
no I meant coin::burn_from

if some other coin type has BurnCapability and calls this, it will emit wrong event.